### PR TITLE
Replace request with got

### PR DESCRIPTION
**Pull request by MTRNord**
As request is deprecated :)

PS my IDE did some reordering of stuff :( So the diff is a bit bigger than it actually should be :/

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -14,15 +14,17 @@ limitations under the License.
 import * as http from "http";
 import * as https from "https";
 import * as fileType from "file-type";
-import { Buffer } from "buffer";
+import {Buffer} from "buffer";
```
Comment (Sorunome) [inline src/util.ts]: what did you do to all the extra spaces on imports D: they were there on purpose!

File: src/util.ts [hunk 2]
```diff
@@ -34,14 +36,14 @@ export interface IMakeUploadFileData {
 }
 
 export class Util {
-	// tslint:disable-next-line no-any
-	public static async DownloadFile(url: string, options: any = {}): Promise<Buffer> {
+	// tslint:disable-next-line:max-line-length
```
Comment (Sorunome) [inline src/util.ts]: Why not break up the line instead?

```ts
public static async DownloadFile<T extends Response | Response["body"]>(
    url: string,
    options: OptionsOfDefaultResponseBody = {},
): Promise<Buffer> {
    // function body

File: src/util.ts [hunk 3]
```diff
@@ -34,14 +36,14 @@ export interface IMakeUploadFileData {
...
+	public static async DownloadFile<T extends Response | Response["body"]>(url: string, options: OptionsOfDefaultResponseBody = {}): Promise<Buffer> {
```
Comment (Sorunome) [inline src/util.ts]: What did you make this a template for? Some protocol implementations already depend on this method....

File: src/util.ts [hunk 4]
```diff
@@ -34,14 +36,14 @@ export interface IMakeUploadFileData {
...
 		if (!options.method) {
 			options.method = "GET";
 		}
 		options.url = url;
-		options.encoding = null;
-		return await request(options);
+		options.encoding = undefined;
+		return got(options).buffer();
```
Comment (Sorunome) [inline src/util.ts]: `return await got(options).buffer();`, else it is not clear where errors are thrown, if they happen

File: src/util.ts [hunk 5]
```diff
@@ -202,7 +204,7 @@ export class Util {
 			}
 			if (newProfile.avatarUrl || newProfile.avatarBuffer) {
 				log.verbose("Uploading avatar...");
-				const { doUpdate: doUpdateAvatar, mxcUrl, hash } = await Util.MaybeUploadFile(uploadFn, newProfile);
+				const {doUpdate: doUpdateAvatar, mxcUrl, hash} = await Util.MaybeUploadFile(uploadFn, newProfile);
```
Comment (Sorunome) [inline src/util.ts]: Please leave the extra spaces

File: src/util.ts [hunk 6]
```diff
@@ -218,7 +220,7 @@ export class Util {
 		if ((newProfile.avatarUrl !== undefined && newProfile.avatarUrl !== null
 			&& newProfile.avatarUrl !== oldProfile.avatarUrl) || newProfile.avatarBuffer) {
 			log.verbose("Uploading avatar...");
-			const { doUpdate: doUpdateAvatar, mxcUrl, hash } = await Util.MaybeUploadFile(uploadFn, newProfile,
+			const {doUpdate: doUpdateAvatar, mxcUrl, hash} = await Util.MaybeUploadFile(uploadFn, newProfile,
```
Comment (Sorunome) [inline src/util.ts]: spaces

File: src/util.ts [hunk 7]
```diff
@@ -242,7 +244,8 @@ export class Util {
 			cmd.stdout.on("data", (data: string) => {
 				databuf += data;
 			});
-			cmd.stdout.on("error", (error) => { }); // disregard
+			cmd.stdout.on("error", (error) => {
+			}); // disregard
```
Comment (Sorunome) [inline src/util.ts]: unnecessary linebreak introduced

File: src/util.ts [hunk 8]
```diff
@@ -256,7 +259,8 @@ export class Util {
 					reject(err);
 				}
 			});
-			cmd.stdin.on("error", (error) => { }); // disregard
+			cmd.stdin.on("error", (error) => {
+			}); // disregard
```
Comment (Sorunome) [inline src/util.ts]: unnecessary linebreak introduced

**Review by Sorunome — CHANGES_REQUESTED** [review]
Thank you for the PR - some style comments and content comments.

In general it is a good idea to *not* change the code style a project already has, when submitting PRs :)

File: src/util.ts (hunk 3 above)
Comment (Sorunome) [reply src/util.ts]: The template is nowhere used in the method and a `Buffer` is always used. Sounds like the template isn't needed.

**Comment by MTRNord** [comment]
Those space issues and linebreaks were introduced through my kinda stupid IDE :) I will revert that!

File: src/util.ts (hunk 3 above)
Comment (MTRNord) [reply src/util.ts]: I used that from the return type that got had but forgot to remove that generic parameter again. (first had it return Response)

File: src/util.ts (hunk 4 above)
Comment (MTRNord) [reply src/util.ts]: 👍

File: src/util.ts [hunk 9]
```diff
@@ -16,13 +16,15 @@ import * as https from "https";
 import * as fileType from "file-type";
 import { Buffer } from "buffer";
 import * as hasha from "hasha";
-import { MatrixClient } from "@sorunome/matrix-bot-sdk";
+import { MatrixClient}  from "@sorunome/matrix-bot-sdk";
```
Comment (Sorunome) [inline src/util.ts]: Space typo?

**Review by Sorunome — APPROVED** [review]
Super tiny nitpicking. ^^ Thanks, though. Will merge either way

File: src/util.ts (hunk 9 above)
Comment (MTRNord) [reply src/util.ts]: yep :D

**Comment by Sorunome** [comment]
Awesome, thank you

**Pull request merged**